lib/src/components/NxCollapsibleItemsSelect/NxCollapsibleRadioSelect/__tests__/NxCollapsibleRadioSelect.test.tsx

```diff
@@ -4,173 +4,495 @@
+        view = quickRender({ onChange, disabled: true }),
+        radios = view.getAllByRole('menuitemradio');
 
-    it('renders unchecked <NxRadio> for all options if selectedId prop is not provided', function () {
-      const shallowRender = getShallow();
-      const renderOption = shallowRender.renderProp('renderOption');
+    expect(onChange).not.toHaveBeenCalled();
+    expect(radios[0]).toBeDisabled();
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('isChecked', false);
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('isChecked', false);
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('isChecked', false);
+    await user.click(radios[0]);
+    expect(onChange).not.toHaveBeenCalled();
+  });
+
+  describe('collapsible items', function() {
+    it('renders options with role=menuitemradio in an element with role=menu', function() {
+      const view = quickRender(),
+          menu = view.getByRole('menu'),
+          options = within(menu).getAllByRole('menuitemradio');
+
+      expect(menu).toBeInTheDocument();
+      expect(options.length).toBe(3);
     });
 
-    it('renders enabled <NxRadio> for all options if component is not disabled', function () {
-      const renderOption = getShallow().renderProp('renderOption');
+    it('renders text content according to option\'s name prop', function() {
+      const options = quickRender().getByRole('menu').children;
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('disabled', false);
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('disabled', false);
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('disabled', false);
+      expect(options[0]).toHaveTextContent('Foo');
+      expect(options[1]).toHaveTextContent('Bar');
+      expect(options[2]).toHaveTextContent('Null');
     });
 
-    it('renders disabled <NxRadio> for all options if component is disabled', function () {
-      const shallowRender = getShallow({disabled: true});
-      const renderOption = shallowRender.renderProp('renderOption');
+    it('renders accessible name according to option\'s name prop', function() {
+      const options = quickRender().getAllByRole('menuitemradio');
 
-      expect(renderOption({id: 'foo', name: 'Foo'})).toHaveProp('disabled', true);
-      expect(renderOption({id: 'bar', name: 'Bar'})).toHaveProp('disabled', true);
-      expect(renderOption({id: null, name: 'Null'})).toHaveProp('disabled', true);
+      expect(options[0]).toHaveAccessibleName('Foo');
+      expect(options[1]).toHaveAccessibleName('Bar');
+      expect(options[2]).toHaveAccessibleName('Null');
     });
 
-    it('uses option name as radio label', function () {
-      const renderOption = getShallow().renderProp('renderOption');
+    it('renders an unchecked radio if option is not selected', function() {
+      expect(quickRender({ selectedId: 'bar' }).getAllByRole('menuitemradio')[0]).not.toBeChecked();
```

Comment:
Cleaner to get by role + name than to get all and then use array indexing

Reply:
> Cleaner to get by role + name than to get all and then use array indexing

Updated here: [cffaa36](https://github.com/sonatype/sonatype-react-shared-components/pull/932/commits/cffaa36db07a65baebd11e5f0c4bb66c1fc75819)